tests/models/speecht5/test_modeling_speecht5.py

```diff
@@ -1024,6 +1031,54 @@ def test_generation(self):
         generated_speech_with_generate = model.generate(input_ids, attention_mask=None)
         self.assertEqual(generated_speech_with_generate.shape, (1820, model.config.num_mel_bins))
 
+    def test_batch_generation(self):
+        model = self.default_model
+        model.to(torch_device)
+        processor = self.default_processor
+        vocoder = self.default_vocoder
+        set_seed(555)  # make deterministic
+
+        input_text = [
+            "mister quilter is the apostle of the middle classes and we are glad to welcome his gospel",
+            "nor is mister quilter's manner less interesting than his matter",
+            "he tells us that at this festive season of the year with christmas and rosebeaf looming before us",
+        ]
+        inputs = processor(text=input_text, padding="max_length", max_length=128, return_tensors="pt").to(torch_device)
+
+        speaker_embeddings = torch.zeros((1, 512), device=torch_device)
+        spectrograms, spectrogram_lengths = model.generate_speech(
+            input_ids=inputs["input_ids"],
+            speaker_embeddings=speaker_embeddings,
+            attention_mask=inputs["attention_mask"],
+        )
+        self.assertEqual(spectrograms.shape, (3, 262, model.config.num_mel_bins))
+        waveforms = vocoder(spectrograms)
+        waveform_lengths = [int(waveforms.size(1) / max(spectrogram_lengths)) * i for i in spectrogram_lengths]
```

Comment:
I am not sure I understand why this is useful? Could be return only if a `return_spectrogram_length`? this way no need for different outputs and BC!

Reply:
Same reason as the above comment for waveforms.